pkg/backup/backup_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backup
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/coreos/etcd-operator/pkg/util"
+)
+
+func TestBackupVersionCompatiblity(t *testing.T) {
+	tests := []struct {
```

Comment:
let's generate the temp data in this test file, and clean it up after test. we should make the test has full context when it is possible. now the reader has to know we have an existing data. if the reader does not look at this exact commit (this commit contains all changes), it is hard for them to know.